samples/star/src/main/kotlin/com/slack/circuit/star/repo/PetRepository.kt

```diff
@@ -44,6 +45,12 @@ class PetRepositoryImpl @Inject constructor(private val petFinderApi: PetfinderA
   }
 
   private suspend fun fetchAnimals() {
-    animals = petFinderApi.animals(limit = 100).animals
+    animals =
+      try {
+        petFinderApi.animals(limit = 100).animals
+      } catch (e: HttpException) {
+        // Sometimes petfinder's API throws 429s for no reason.
```

Comment:
Opportunistic as this was happening to me while I was working on this. Maybe in the future we wanna show some sort of error state